src/common/string/filter/sequence-matching.ts

```diff
@@ -64,21 +80,72 @@ export const fuzzySequentialMatch = (
 export interface ScorableTextItem {
   score?: number;
   strings: string[];
+  decoratedStrings?: string[][];
 }
 
 type FuzzyFilterSort = <T extends ScorableTextItem>(
   filter: string,
-  items: T[]
+  items: T[],
+  decorate?: MatchDecorator
 ) => T[];
 
-export const fuzzyFilterSort: FuzzyFilterSort = (filter, items) => {
+export const fuzzyFilterSort: FuzzyFilterSort = (
+  filter,
+  items,
+  decorate = createMatchDecorator("[", "]")
+) => {
   return items
     .map((item) => {
-      item.score = fuzzySequentialMatch(filter, item);
+      const match = fuzzySequentialMatch(filter, item, decorate);
+
+      item.score = match?.score;
+      item.decoratedStrings = match?.decoratedStrings;
+
       return item;
     })
     .filter((item) => item.score !== undefined)
     .sort(({ score: scoreA = 0 }, { score: scoreB = 0 }) =>
       scoreA > scoreB ? -1 : scoreA < scoreB ? 1 : 0
     );
 };
+
+type MatchDecorator = (word: string, scores?: FuzzyScore) => string[];
+export const createMatchDecorator: (
+  left: string,
+  right: string
+) => MatchDecorator = (left, right) => (word, scores) =>
+  _decorateMatch(word, [left, right], scores);
+
+const _decorateMatch: (
+  word: string,
+  surroundWith: [string, string],
```

Comment:
Instead of a string, pass it and html tagged literal, so we don't have to do `unsafeHtml`.

Reply:
How about `string | TemplateResult`?

Reasons being that:
1. The test throws an exception if I try to import `html` into either `sequence_matching.test.ts` or `sequence-matching.ts`
2. Allowing `string` makes it easier to write unit tests for highlighting, and allows for debugging to console log
3. Future components may not want to inject htm